### ISSUE TITLE
Guest count plummeting for no reason

### ISSUE THREAD

**Issue by Zygoat23** [OpenRCT2/OpenRCT2#18310]
### Operating System

Windows 10, 64 bit

### OpenRCT2 build

OpenRCT2 v0.4.2

### Describe the issue

I've seen this in action in two different scenarios where the guest goal was high. In both scenarios, I had a soft guest cap that was above the goal calculated by the soft guest cap calculator plugin. The guest count should have been continuously increasing as a result, but instead and in both cases, at about 1000 guests below the goal and guest cap, it started dropping rapidly. It wouldn't stop dropping either. Both parks had around 999 park rating, so a low park rating wasn't the cause. The first scenario it happened required 5000 guests and had normal settings. The second scenario required 4000 guests and had harder guest generation turned on. This eliminates that setting as a cause for the bug, since it happened regardless of its activation. The rate at which guests leave is about the same rate that guests arrive using ads, so ads are not very useful in countering this bug. 

![Fabius Flatlands 2022-10-12 15-25-35](https://user-images.githubusercontent.com/62668625/195444919-606e4317-c52b-4599-9357-25e6f60f2cb2.png)

### Area(s) with issue?

New Save Format (NSF)

### Steps to reproduce

1. Open the attached park file
2. Observe the guest count drop

### Attachments

[Fabius Flatlands.zip](https://github.com/OpenRCT2/OpenRCT2/files/9768682/Fabius.Flatlands.zip)

**Comment by ocalhoun6** [comment]
Do you have ATMs, restrooms, food and drink stalls, etc, etc, etc to make sure the guests are happy, have plenty of money, and have no reason to leave?

Perhaps it starts dropping rapidly because a while ago it increased rapidly, and now that large group of guests who all came in about the same time are running out of money (or some other issue) and all deciding to leave around the same time as well?

-----

Overcrowding can also cause peeps to decide to leave the park. Are parts of your park getting overcrowded when the guest count becomes high enough?

**Comment by Zygoat23** [comment]
If the cause is any of the things you mentioned, the only thing it could be is overcrowding but I've never had that issue before in any of the vanilla RCT1 and 2 scenarios, even Amity Airfield and Extreme Heights

**Comment by krrg** [comment]
Sorry to revive an older issue, but I came across this while investigating another possible bug on guest generation, and at least in the scenario you attached, the problem boiled down to overcrowding and guests running out of money. The numbers coming out of the guest generation code all looked very normal.

I added 3 more ATMs, did the absolute bare minimum of fixes to the path network, doubled the number of handymen and mechanics (just in case), put it into fastest of fast forward modes, and the scenario completed successfully with a couple months to spare and a healthy bank balance.

![image](https://github.com/OpenRCT2/OpenRCT2/assets/8496492/0ef316e3-c612-4f11-aedb-78d38367c01c)
